pandas/core/groupby/groupby.py

```diff
@@ -2625,24 +2625,39 @@ def post_processor(vals: np.ndarray, inference: np.dtype | None) -> np.ndarray:
             libgroupby.group_quantile, labels=ids, qs=qs, interpolation=interpolation
         )
 
+        # Put '-1' (NaN) labels as the last group so it does not interfere
+        # with the calculations. Note: length check avoids failure on empty
+        # labels. In that case, the value doesn't matter
+        na_label_for_sorting = ids.max() + 1 if len(ids) > 0 else 0
+        labels_for_lexsort = np.where(ids == -1, na_label_for_sorting, ids)
+
         def blk_func(values: ArrayLike) -> ArrayLike:
             mask = isna(values)
             vals, inference = pre_processor(values)
 
             ncols = 1
             if vals.ndim == 2:
                 ncols = vals.shape[0]
+                shaped_labels = np.broadcast_to(
```

Comment:
why is the broadcast necessary here?

Reply:
The problem was labels is 1-dimensional, but values can be 2. I couldn't figure out a way to make `lexsort` naturally broadcast in that case. I think this broadcast is relatively cheap, though, since `np.broadcast_to` uses views, so it shouldn't increase memory usage (and this op doesn't show up in a profile)

Reply:
The broadcast itself is definitely cheap.  In some cases I've later found that operations done on these broadcasted ndarrays are surprisingly slow.  But if you've profiled it and its not an issue, then no complaints here.

Reply:
Oh, interesting. The profile showed that all time is spent in lexsort, so definitely possible slow access on the broadcasted array is slowing it down. Will check how lexsort timing compares with this broadcast vs a contiguous array

Reply:
Adding a copy after the broadcast doesn't seem to affect lexsort time in a profile